pkg/phlaredb/block_querier.go

```diff
@@ -159,6 +159,23 @@ func (b *BlockQuerier) BlockMetas(ctx context.Context) (metas []*block.Meta, _ e
 	return metas[0 : pos+1], nil
 }
 
+func (b *BlockQuerier) BlockMeta(ctx context.Context, name string) (meta *block.Meta, _ error) {
```

Comment:
I had to introduce a direct way to read a single meta, otherwise, using `BlockMetas` with load every block in the bucket (potentially thousands of them)